app/src/main/java/com/android/sample/ui/fridge/FridgeScreen.kt

```diff
@@ -30,29 +117,414 @@ import com.android.sample.ui.utils.PlateSwipeScaffold
+                          chunk.getOrNull(0)?.let { card1 ->
+                            ItemCard(Modifier.weight(1f), card1, userViewModel)
+                          }
+
+                          // Second card in the chunk (if present)
+                          chunk.getOrNull(1)?.let { card2 ->
+                            ItemCard(Modifier.weight(1f), card2, userViewModel)
+                          }
+
+                          // Add an empty spacer if only one card exists in the chunk
+                          if (chunk.size == 1) {
+                            Spacer(modifier = Modifier.weight(1f).padding(PADDING_16.dp))
+                          }
+                        }
+                  }
+            }
+          }
+        }
+    // button to add ingredient
+    PlateSwipeButton(
+        stringResource(R.string.add_ingredient),
+        modifier = Modifier.padding(PADDING_16.dp).align(Alignment.BottomCenter),
+        onClick = {
+          userViewModel.clearIngredientList()
+          userViewModel.clearSearchingIngredientList()
+          userViewModel.clearIngredient()
+          navigationActions.navigateTo(Screen.FRIDGE_SEARCH_ITEM)
+        })
+  }
+}
+
+/**
+ * Item Card
+ *
+ * @param modifier
+ * @param card
+ * @param userViewModel
+ */
+@SuppressLint("AutoboxingStateCreation")
 @Composable
-fun FridgeContent(paddingValues: PaddingValues) {
+private fun ItemCard(
+    modifier: Modifier,
+    card: Pair<FridgeItem, Ingredient>,
+    userViewModel: UserViewModel
+) {
+  var showEditDialog by remember { mutableStateOf(false) }
+  var updatedQuantity by remember { mutableIntStateOf(card.first.quantity) }
+
   Column(
-      modifier = Modifier.fillMaxSize().padding(paddingValues),
-      verticalArrangement = Arrangement.Center,
-      horizontalAlignment = Alignment.CenterHorizontally) {
+      modifier = modifier.padding(PADDING_8.dp),
+      horizontalAlignment = Alignment.CenterHorizontally,
+      verticalArrangement = Arrangement.Center) {
+        Card(
+            modifier = Modifier.padding(PADDING_8.dp),
+            colors = CardDefaults.cardColors(MaterialTheme.colorScheme.secondary),
+            shape = RoundedCornerShape(CARD_BORDER_ROUND.dp),
+            elevation = CardDefaults.cardElevation(defaultElevation = CARD_ELEVATION.dp)) {
+              Box(modifier = Modifier.fillMaxSize()) {
+                // Pencil Icon - Positioned at Top Left
+                IconButton(
+                    onClick = { showEditDialog = true },
+                    modifier =
+                        Modifier.align(Alignment.TopEnd)
+                            .padding(PADDING_8.dp)
+                            .size(EDIT_ICON_SIZE.dp),
+                ) {
+                  Icon(
+                      imageVector = Icons.Default.Edit,
+                      contentDescription = "Edit ${card.second.name} Quantity",
+                      tint = MaterialTheme.colorScheme.onSecondary,
+                  )
```

Comment:
Nitpick: I think it would be great if we allowed the user to directly edit the card when they click on it, instead of using an icon. To me, this feels more natural and fluid. What do you think? 

Here is an [example](https://stackoverflow.com/questions/77045138/clickable-modifier-doesnt-work-with-card-in-jetpack-compose).

Reply:
I thought about it and decided to use an edit button. I believe a visual element like an icon makes it easier for users to understand how to edit an ingredient in the fridge. Without a tutorial, it might not be clear that clicking on an item deletes it. I’ve also get comments on the swipe page about how some actions (e.g., which swipe does what or how to access recipes) are not fully clear. For this feature, I opted for a clearer approach. I'll gather user feedback and make updates next week if needed. Does this work for you?

Reply:
**Yes, this is perfect.** As you mentioned, my feedback reflects the perspective of a specific type of user, someone with extensive experience using mobile apps frequently. For other users, such as older individuals, the design or functionality might not be as intuitive. Conducting user testing to gather insights from a broader audience would be an excellent approach to ensure inclusivity and usability for everyone.